src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1021,6 +1028,35 @@ protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator
 
             _numberOfAllocatedStringsValues = 0;
 
+            if (releaseAllocatedStringValues && _allocatedStringValuesCount >= _maxNumberOfAllocatedStringValues / 2)
+            {
+                var now = DateTime.UtcNow;
+                if (now - _lastAllocatedStringValueTime >= _allocatedStringValuesCleanInterval)
+                {
+                    _lastAllocatedStringValueTime = now;
+
+                    var index = _allocatedStringValuesCount;
+                    for (var i = 0; i < 4; i++)
+                    {
+                        index /= 2;
+                        var item = _allocateStringValues[index];
+                        if (item != null)
+                            break;
+                    }
+
+                    for (var i = index; i < _allocatedStringValuesCount; i++)
+                    {
+                        var item = _allocateStringValues[i];
+                        if (item == null)
+                            break;
+
+                        _allocateStringValues[i] = null;
+                    }
```

Comment:
What are you trying to do here?

Reply:
I'm releasing the LSV instances (empty ones) so GC can collect them.

Reply:
That is why we have `Array.Clear`, no? Why are you doing this one at a time?
And why are you probing the array?

Reply:
Done